internal/pkg/manifest/svc.go

```diff
@@ -300,12 +314,48 @@ func (a *Count) UnmarshalYAML(unmarshal func(interface{}) error) error {
 // Autoscaling represents the configurable options for Auto Scaling.
 type Autoscaling struct {
 	Range        Range          `yaml:"range"`
-	CPU          *string        `yaml:"cpu"`
-	Memory       *string        `yaml:"memory"`
+	CPU          *Percentage    `yaml:"cpu"`
+	Memory       *Percentage    `yaml:"memory"`
 	Requests     *int           `yaml:"requests"`
 	ResponseTime *time.Duration `yaml:"response_time"`
 }
 
+// Options converts the service's Auto Scaling configuration into a format parsable
+// by the templates pkg.
+func (a *Autoscaling) Options() (*template.AutoscalingOpts, error) {
```

Comment:
💭 You know I think I made a mistake by having the transformation done within the `manifest` pkg to `template` objects. 
It shouldn't be the responsibility of the producer package to provide a way of transforming its data for all of its consumers. Instead, the other way around makes more sense.

So for example: `template.TransformManifestAutoscaling(asg manifest.Autoscaling) AutoscalingOpts`

I'm not saying you should do this refactor here if it's a lot to do, but I'm curious to hear your thoughts about it and how much work that would be.

Reply:
Yeah that sounds good to me. Actually there is not a lot of work to do the refactoring so i'll do it in this PR!

Reply:
take back my words. it is not easy since there will be cyclic reference issue because our manifest pkg currently depends on template. I won't do the refactoring here.